src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -948,10 +946,9 @@ VideoDecoder::AVFrameStream VideoDecoder::decodeAVFrame(
 
   resetDecodeStats();
 
-  // Seek if needed.
-  if (desiredPtsSeconds_.has_value()) {
+  if (cursorWasJustSet_) {
     maybeSeekToBeforeDesiredPts();
-    desiredPtsSeconds_ = std::nullopt;
+    cursorWasJustSet_ = false;
```

Comment:
I'm not particularly fond of this logic, but don't shoot the messenger. This PR is only making an existing stateful logic a little bit more obvious. I.e. now we have an obvious `bool` field, while before we had a more confusing `std::optional` logic.

We could try to simplify this further, I gave it a quick try but our tests failed in some weird edge-cases, so I'm leaving this for later.